filter/patterns_storage_test.go

```diff
@@ -46,20 +49,73 @@ func TestProcessIncomingMetric(t *testing.T) {
 
 	Convey("When valid non-matching metric arrives", t, func() {
 		patternsStorage.metrics = metrics.ConfigureFilterMetrics(metrics.NewDummyRegistry())
-		matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("disk.used 12 1234567890"))
-		So(matchedMetrics, ShouldBeNil)
-		So(patternsStorage.metrics.TotalMetricsReceived.Count(), ShouldEqual, 1)
-		So(patternsStorage.metrics.ValidMetricsReceived.Count(), ShouldEqual, 1)
-		So(patternsStorage.metrics.MatchingMetricsReceived.Count(), ShouldEqual, 0)
+		Convey("For plain metric", func() {
+			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("disk.used 12 1234567890"))
+			So(matchedMetrics, ShouldBeNil)
+			So(patternsStorage.metrics.TotalMetricsReceived.Count(), ShouldEqual, 1)
+			So(patternsStorage.metrics.ValidMetricsReceived.Count(), ShouldEqual, 1)
+			So(patternsStorage.metrics.MatchingMetricsReceived.Count(), ShouldEqual, 0)
+		})
+
+		Convey("For tag metric", func() {
+			matchedMetrics := patternsStorage.ProcessIncomingMetric([]byte("disk.used;tag1=val1 12 1234567890"))
+			So(matchedMetrics, ShouldBeNil)
+			So(patternsStorage.metrics.TotalMetricsReceived.Count(), ShouldEqual, 1)
+			So(patternsStorage.metrics.ValidMetricsReceived.Count(), ShouldEqual, 1)
+			So(patternsStorage.metrics.MatchingMetricsReceived.Count(), ShouldEqual, 0)
+		})
+
+		Convey("For plain metric which is like tag metric", func() {
```

Comment:
The name of test case is difficult for understand (for me). May be it would be better to call it like "For plain metric which matches with tag in trigger pattern"?